src/Attribute.php

```diff
@@ -0,0 +1,159 @@
+use Yiisoft\Translator\TranslatorInterface;
+use Yiisoft\Arrays\ArrayHelper;
+use Yiisoft\Html\Html;
+
+final class Attribute
+{
+    private string $name = '';
+    private ?string $label = null;
+    /** @var Closure|string|null */
+    private $format;
+    private bool $encode = false;
+    private TranslatorInterface $translator ;
+    /** @var Closure|Stringable|string|null */
+    private $value;
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    public function name(string $name): self
+    {
+        if (!preg_match('/^(?P<name>[^:]+)(:(?P<format>\w*))?(:(?P<label>.*))?$/', $name, $matches)) {
+            throw new InvalidConfigException(
+                'The attribute must be specified in the format of "attribute", "attribute:format" or "attribute:format:label"'
+            );
+        }
+
+        $this->name = $matches['name'];
+
+        if (isset($matches['format'])) {
+            $this->format('{0,' . $matches['format'] . '}');
+        }
+
+        if (isset($matches['label'])) {
+            $this->label($matches['label']);
+        }
+
+        return $this;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function label(?string $label): self
+    {
+        $this->label = $label;
+
+        return $this;
+    }
+
+    public function getLabel(): ?string
+    {
+        return $this->label;
+    }
+
+    public function encode(bool $encode): self
+    {
+        $this->encode = $encode;
+
+        return $this;
+    }
+
+    /**
+     * @param Closure|string|null $format
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return self
+     */
+    public function format($format): self
+    {
+        if ($format !== null && !is_string($format) && !is_a($format, Closure::class)) {
+            throw new InvalidArgumentException('Format must be type of "null", "string" or Closure instance');
+        }
+        /** @var Closure|string|null $format */
+        $this->format = $format;
+
+        return $this;
+    }
+
+    /**
+     * @param Closure|Stringable|string|null $value
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return self
+     */
+    public function value($value): self
+    {
+        if ($value !== null && !is_string($value) && !is_a($value, Closure::class) && !is_a($value, Stringable::class)) {
+            throw new InvalidArgumentException('Value must be type of "null", "string" or Closure instance');
+        }
+        /** @var Closure|Stringable|string|null $value */
+        $this->value = $value;
+
+        return $this;
+    }
+
+    /**
+     * @param array|object $model
+     *
+     * @throws RuntimeException
+     * @throws InvalidArgumentException
+     *
+     * @return string
+     */
+    public function getValue($model): string
+    {   /** @psalm-suppress DocblockTypeContradiction */
```

Comment:
Why psalm-suppress

Reply:
> Why psalm-suppress

I don't know how legal fix it. I was try

Reply:
I'll check it, I'll try to fix it

Reply:
> I'll check it, I'll try to fix it

Thx

Reply:
```suggestion
    /**
     * @param mixed $model
     *
     * @throws RuntimeException
     * @throws InvalidArgumentException
     *
     * @return string
     */
```
You are telling the psalm that it is array or object, so if you validate that it is not array, then it is object, so since it can be any type since it does not have typehint, define it mixed.